tests/functional/test_workflow.py

```diff
@@ -1243,7 +1243,7 @@ def mock_tmp_input(requests_mock):
             )
             # check that all expected files made it through the listing/directory input/output chaining between steps
             output_files = "\n".join(os.path.join(*line.rsplit("/", 2)[-2:]) for line in output_lines)
-            expect_files = "\n".join(os.path.join("input_dir", os.path.split(file)[-1]) for file in expect_http_files)
+            expect_files = "\n".join(os.path.join("output_dir", os.path.split(file)[-1]) for file in expect_http_files)
```

Comment:
Just to make sure I have interpreted the reason of this change correctly. 

After passing the `merge` workflow step, the produced directory URL is something like `http://localhost/wps-outputs/{merge-jobID}/output_dir/`, with all expected files on disk under somewhere like `/tmp/wps_outputs/{merge-jobID}/output_dir/`. 

When the `listing` step receives this directory URL as input, isn't it supposed to stage it under the relevant input ID, some something like `/tmp/wps_outputs/{listing-jobID}/input_dir/output_dir/`? Originally, the last `output_dir/` part was not reused, hence the expected `input_dir` as the last part. 

If my interpretation is correct, there is probably a need to perform another check here to ensure `input_dir/output_dir/` is the full directory path. I guess it is not shown in the output text file because the `ls` call is relative to `output_dir/` only.

Reply:
Haven't added this last assert yet, will get to it next

Reply:
However, I looked at it quickly; what seems to happen it that `input_dir` is being replaced by `output_dir`, not adding to it. It makes sense, kind of, since the "input" is "output_dir" (https://github.com/crim-ca/weaver/blob/master/tests/functional/application-packages/WorkflowDirectoryListing/package.cwl#L20), but I don't see why it's happening yet. 

I'll have to look more into it.

Reply:
Yes. 
Process `merge` should produce `/tmp/wps_outputs/{merge-jobID}/output_dir/`.
With your addition, the last part `output_dir/` is preserved.
The original behaviour always nests under `<input-id>` to ensure no conflict, so `input_dir` for the `listing` process. 
The resulting fetched location is `/tmp/<cwl-stage-tmp>/input_dir/output_dir/`. 
So far, everything as expected.

With a quick test of `cwltool` with https://github.com/crim-ca/weaver/blob/master/tests/functional/application-packages/DirectoryListingProcess/package.cwl, the output contains entries like:
`/var/lib/cwl/stg9afec70c-3a16-49a3-946c-1039b5c07095/output_dir/...`

I've tried the same `listing` CWL with an extra dir input, and `cwltool` itself attributes different `stg<rand>` for each input.
Therefore, it replicates the goal of the distinct directories per `<input-id>`. 
I think it would be safe to consider this test valid as is.